executor/adapter.go

```diff
@@ -1095,9 +1095,6 @@ func getEncodedPlan(sctx sessionctx.Context, p plannercore.Plan, genHint bool, n
 	}
 	if genHint {
 		hints := plannercore.GenHintsFromPhysicalPlan(p)
-		if n != nil {
-			hints = append(hints, hint.ExtractTableHintsFromStmtNode(n, nil)...)
-		}
```

Comment:
Why should we remove these lines?

Reply:
I think it is enough to gather hints from the physical plan .

Reply:
Please add some tests about table hints in the CTE statement. @zhaoxugang